### DISCUSSION TITLE
Very close trials in Module 3, but not accepted by autograder

### DISCUSSION THREAD

**Discussion by kenzo-aspuru-takata** [AccelerationConsortium/ac-microcourses#25] [Hello World] [answered]
My stuff is only off by about 0.5 ~0.2% and it says in the debug console the test will not be exactly the same.

Something said in the output:

> Random seed set to 42. Note that this setting only affects the Sobol quasi-random generator and BoTorch-powered Bayesian optimization models. For the latter models, setting random seed to the same number for two optimizations will make the generated trials similar, but not exactly the same, and over time the trials will diverge more.

The error:

```python
AssertionError: Expected first suggested experiment issued to run_color_experiment to be approximately 

{'R': 255, 
'G': 26, 
'B': 210, 
'ch410': 219.33087856963076, 
'ch440': 248.248387805653, 
'ch470': 282.9731413261822, 
'ch510': 268.15146335347225, 
'ch550': 239.06822906707887, 
'ch583': 240.12586524637015, 
'ch620': 273.09731873935584, 
'ch670': 237.84121011577813}

got

{'R': 254.36590522527695,
 'G': 26.613429933786392, 
'B': 209.8596078157425, 
'ch410': 219.26391480929513, 
'ch440': 248.22841161673279, 
'ch470': 283.04441855109593, 
'ch510': 268.31622424188777, 
'ch550': 239.31004096667493, 
'ch583': 240.21734078897398, 
'ch620': 272.7421011488162, 
'ch670': 237.3737462995075}
```

Code:
https://github.com/ACC-HelloWorld/3-bayesian-optimization-kenzo-aspuru-takata/blob/main/bayesian_optimization.py.

log:

[output (1).txt](https://github.com/user-attachments/files/16286851/output.1.txt)

**Comment by sgbaird** [comment, marked as answer]
Hi @kenzo-aspuru-takata, it looks like you might be optimizing over a continuous space. If you specify a float for the bounds (e.g., `0.0`) Ax interprets this as a continuous parameter, but the RGB values are meant to be integer values. If you specify an integer (e.g., `0`) Ax will interpret this as an integer parameter. The type can also be specified explicitly, but the assignment README doesn't go over that.

**Reply by kenzo-aspuru-takata** [reply]
That was indeed the problem! I entirely forgot that RGB values are integers. It was also a bit confusing because the "test_parameters_and_objectives" test was succeeding, but not the "test_calls" test, so I didn't think to check my parameters.
